### PR TITLE
Fix console warning due to uncontrolled SelectWidget

### LINKED ISSUE #1720: Warning about uncontrolled input component on select widget
Steps to reproduce:

- hard refresh volto
- trigger add form by selection "create folder"
- in the "Settings" tab, set "Allow discussion" to "yes"

```
Warning: A component is changing an uncontrolled input of type hidden to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
    in input (created by Select)
    in div (created by Context.Consumer)
    in EmotionCssPropInternal (created by SelectContainer)
    in SelectContainer (created by Select)
    in Select (created by StateManager)
    in StateManager (created by InnerLoadable)
    in InnerLoadable (created by Context.Consumer)
    in Unknown (created by ForwardRef)
    in ForwardRef (at SelectWidget.jsx:319)
    in div (created by GridColumn)
    in GridColumn (at FormFieldWrapper.jsx:106)
    in div (created by GridRow)
    in GridRow (at FormFieldWrapper.jsx:90)
    in div (created by Grid)
    in Grid (at FormFieldWrapper.jsx:89)
    in div (created by FormField)
    in FormField (at FormFieldWrapper.jsx:82)
    in FormFieldWrapper (at SelectWidget.jsx:279)
    in SelectWidget (created by ConnectFunction)
    in ConnectFunction (created by Context.Consumer)
    in injectIntl(Connect(SelectWidget)) (at Field.jsx:172)
    in Field (created by Context.Consumer)
    in injectIntl(Field) (at Form.jsx:902)
    in div (created by Tab)
    in Tab (at Form.jsx:885)
    in div (created by SegmentGroup)
    in SegmentGroup (at Form.jsx:883)
    in form (created by Form)
    in Form (at Form.jsx:878)
    in div (created by Container)
    in Container (at Form.jsx:877)
    in Form (at FormContext.jsx:59)
    in FormStateProvider (at FormContext.jsx:55)
    in FormWithState (at FormContext.jsx:73)
    in ForwardRef (created by Context.Consumer)
    in injectIntl(Component) (created by ForwardRef)
    in ForwardRef (at Add.jsx:241)
    in div (at Add.jsx:235)
    in Add (created by ConnectFunction)
    in ConnectFunction (created by Context.Consumer)
    in injectIntl(Connect(Add)) (created by 
```

### PR DESCRIPTION
**Pull request by nileshgulia1**
fixes #1720

### CODE REVIEW

**Review by tiberiuichim — APPROVED** [review]
While reviewing this PR I've noticed some strange behavior from Volto and plone.restapi.

When toggling the discussion field value, Plone seems to "refuse" to change that value.  I can see in the PATCH that allowed_discussion is set to "True", but the value change doesn't seem to stick, it is returned from the server as "false". Maybe the vocabulary is wrong?

File: src/components/manage/Widgets/SelectWidget.jsx [hunk 1]
```diff
@@ -96,6 +96,9 @@ function getDefaultValues(choices, value) {
       value: 'no-value',
     };
   }
+  if (value === null) {
```
Comment (tiberiuichim) [inline src/components/manage/Widgets/SelectWidget.jsx:99]: There's no need for this code, same is achieved if you remove the else return {} at the end on this function.

**Review by tiberiuichim — CHANGES_REQUESTED** [review]

**Comment by tiberiuichim** [comment]
The fix is nice, it makes the placeholder option work instead of just showing an empty select.

**Comment by nileshgulia1** [comment]
@tiberiuichim Nice catch! I'm trying to figure out the same why the default value of allow-discussion is setting to false/No everytime.

**Comment by tiberiuichim** [comment]
```
function getDefaultValues(choices, value) {
  if (!isObject(value) && isBoolean(value)) {
    // We have a boolean value, which means we need to provide a "No value"
    // option
    const label = find(choices, (o) => getBoolean(o[0]) === value);
    return label
      ? {
          label: label[1],
          value,
        }
      : {};
  }
  if (value === 'no-value') {
    return {
      label: this.props.intl.formatMessage(messages.no_value),
      value: 'no-value',
    };
  }
  if (isObject(value)) {
    return {
      label: value.title !== 'None' && value.title ? value.title : value.token,
      value: value.token,
    };
  }
  if (value && choices.length > 0) {
    return { label: find(choices, (o) => o[0] === value)[1], value };
  }
}
```

File: src/components/manage/Widgets/SelectWidget.jsx (hunk 1 above)
Comment (nileshgulia1) [reply src/components/manage/Widgets/SelectWidget.jsx]: Done!

**Review by tiberiuichim — APPROVED** [review]

**Pull request merged**